dev/protos/generate.sh

```diff
@@ -94,10 +94,65 @@ function generate_health_service {
   generate_grpc "$proto" "$(dirname "$proto")" "$output" "Visibility=Package" "Client=true" "Server=true" "UseAccessLevelOnImports=true"
 }
 
+function generate_reflection_service {
+  local proto="$here/upstream/grpc/reflection/v1/reflection.proto"
+  local output="$root/Sources/GRPCReflectionService/Generated"
+
+  # Messages were accidentally leaked into public API, they shouldn't be but we
+  # can't undo that change until the next major version.
+  generate_message "$proto" "$(dirname "$proto")" "$output" "Visibility=Package" "UseAccessLevelOnImports=true"
+  generate_grpc "$proto" "$(dirname "$proto")" "$output" "Visibility=Package" "UseAccessLevelOnImports=true"
+}
+
+#- TEST DATA ------------------------------------------------------------------
+
+function generate_reflection_service_descriptor_set {
+  local proto="$here/upstream/grpc/reflection/v1/reflection.proto"
+  local proto_path="$here/upstream"
+  local output="$root/Tests/GRPCReflectionServiceTests/Generated/DescriptorSets/reflection.pb"
+
+  invoke_protoc --descriptor_set_out="$output" "$proto" -I "$proto_path" \
+    --include_source_info \
+    --include_imports
+}
+
+function generate_health_service_descriptor_set {
+  local proto="$here/upstream/grpc/health/v1/health.proto"
+  local proto_path="$here/upstream"
+  local output="$root/Tests/GRPCReflectionServiceTests/Generated/DescriptorSets/health.pb"
+
+  invoke_protoc --descriptor_set_out="$output" "$proto" -I "$proto_path" \
+    --include_source_info \
+    --include_imports
+}
+
+function generate_extended_message_descriptor_set {
```

Comment:
Should this be `base_message`?
```suggestion
function generate_base_message_descriptor_set {
```